### PR TITLE
Fix setting the invalid losses to "inf"

### PR DESCRIPTION
**Pull request by SonyPony**
The propagation of values was wrong.
E.g. loss = [1, 2, NaN, 3, NaN] -> isnan(loss) == [False, False, True, False, True]

Thus loss[isnan(loss)] would have shape (2, ), however isnan(loss) would have shape (5, ).

PyTorch supports propagating a scaler into tensor.

### CODE REVIEW

**Comment by SonyPony** [comment]
I suppose, this is wrong. Should all NaN losses be changed to inf?

image

**Review by astanziola — APPROVED** [review]
